docs/writing/posts/lgtmk.md

```diff
@@ -0,0 +1,229 @@
+
+### Normalized Discounted Cumulative Gain (NDCG) @ K
+
+A nuanced measure that evaluates both the presence and graded relevance of documents, rewarding systems that present the most valuable information first.
+
+$$
+NDCG@K = \frac{DCG@K}{IDCG@K}
+$$
+
+!!! tip "What the fuck is even that?"
+
+    Honestly, I wouldn't worry about it too much, especially in the context of retrieval or generation. If you want to learn more, check out [this great resource](https://www.evidentlyai.com/ranking-metrics/ndcg-metric).
+
+    The TLDR I want to give you here is that this is just a more holistic measure of how well things are being ranked. It's not as aggressive as MRR.
+
+    !!! note "Aggressive?"
+
+        It's my belief that MRR and how it pushes certain rankings to the top is likely responsible for various kinds of echo chambers that might result in recommendation systems. For example, if you're watching a conspiracy theory video, the next thing you'll probably watch is going to be a conspiracy theory video.
+
+## How to improve
+
+Once you have a system in place and some metrics you want to improve, again, the steps are very simple.
+
+1. Choose a metric that aligns with your goals. Distinguish between primary metrics (that must improve) and guardrail metrics (that must not regress).
+2. Formulate a hypothesis and adjust the system.
+3. Evaluate the impact on your chosen metric.
+4. Look at poorly performing examples, and iterate.
+5. Go back to step 2.
+
+!!! warning "Beware of [Simpson's Paradox](https://en.wikipedia.org/wiki/Simpson%27s_paradox)"
+
+    > A paradox in which a trend that appears in different groups of data disappears when these groups are combined, and the reverse trend appears for the aggregate data.
+
+    It's very likely that you might improve the system for one type of query and make it worse for another.
+    To avoid doing this on some level, we can do the following:
+
+    1. Cluster the data (e.g., by query type, data source, etc.).
+    2. Determine if the metric is consistent across different clusters.
+    3. If it is, consider building a router to conditionally select one implementation over another.
+
+## Metrics lead to Business Outcomes
+
+All of these metrics must ultimately be in service of something else. By improving things like precision, recall, and relevancy, what we're really hoping to do is generate better results for the business. The question then you have to ask yourself is, "What does that actually improve?". Here are a couple of things that you might want to consider.
+
+1. **User Satisfaction**: Are users happy with the answers they're getting? Could be defined by Thumb Up/Down or NPS.
+2. **Engagement**: Are users coming back to the platform? Are they spending more time on the platform?
+3. **Conversion**: Are users buying more things? Are they clicking on more ads?
+4. **Retention**: Are users staying on the platform longer? Are they coming back more often? Do we want to improve time spent?
```

Comment:
reducing time spent on the platform?  or increase number of distinct queries per same unit of time measurement? as a proxy for productivity boost?

Reply:
the better the search the short time user will use the system.